### ISSUE TITLE
Window `move` does't preserve its layout (unlike i3)

### ISSUE THREAD

**Issue by 1ace** [swaywm/sway#4038]
Sway version: 1.0

Steps to reproduce:
- `layout tabbed`
- open 3 windows
- on the right-most window: `move right`
- open 1 more window, while still on the 3rd window (the one moved to the right)

Expected result (ie. `i3` behaviour):
- first two windows are on the left half of the screen, tabbed
- 4th window is on the right half of the screen, tabbed over the 3rd window

Actual result:
- first two windows are on the left third of the screen, tabbed _(good behaviour)_
- 3rd window in the middle third of the screen
- 4th window in the right third of the screen
- layout has changed to `splith`

**Comment by hmohamad** [comment]
Can confirm this problem. Sway seems to switch to `splith` automatically, whereas in i3 new windows inherit the layout of their parents.

Pretty annoying, because my workflow used to be stacking per default (`via worksace_layout stacking`) and then manually splitting by moving the windows out of the stacking container.

**Comment by fdlamotte** [comment]
Thanks for addressing this issue.

I've tested the fix proposed in recent commits, it works  when playing the scenario proposed in the pull request (from a new workspace). But then I experienced some cases where it does not work 
 - It doesn't work if there are one or two tabs
 - If I open a terminal, give it a tabbed layout, open two other terminals and move one of the terminals out of the container, it works as expected (two containers with tabbed layout). If I move back the terminal in the first container, it enters in the tabbed layout as expected. But now if I try to move the terminal out of the container again, it looses the tabbed layout.
 - If I open a terminal and give its container a tabbed layout, open three other terminals, move one at the left and another at the bottom or at the right, the second terminal looses the tabbed layout
